libtcmu.c

```diff
@@ -966,20 +1022,23 @@ struct tcmulib_cmd *tcmulib_get_next_command(struct tcmu_device *dev)
 			struct tcmulib_cmd *cmd;
 			uint8_t *cdb = (uint8_t *) mb + ent->req.cdb_off;
 			int cdb_len = tcmu_get_cdb_length(cdb);
+			uint16_t cmd_id = ent->hdr.cmd_id;
 
 			if (cdb_len < 0) {
-				/*
-				 * This should never happen so just drop cmd
-				 * for now instead of adding a lock in the
-				 * main IO path.
-				 */
+				tcmulib_entry_complete(dev, cmd_id, NULL,
+						       TCMU_NOT_HANDLED);
+				completed = true;
 				break;
 			}
 
 			/* Alloc memory for cmd itself, iovec and cdb */
 			cmd = malloc(sizeof(*cmd) + sizeof(*cmd->iovec) * ent->req.iov_cnt + cdb_len);
-			if (!cmd)
-				return NULL;
+			if (!cmd) {
+				tcmulib_entry_complete(dev, cmd_id, NULL,
+						       TCMU_NOT_HANDLED);
```

Comment:
Shouldn't we return to the initiator the  key=HARDWARE_ERROR  and ascq=ASC_INTERNAL_TARGET_FAILURE? or something like that ?
The default will be : 
key ==  illegal request
ascq == invalid command operation code

Reply:
You are right. Will fix. Thanks for reviewing.